datafusion/optimizer/src/common_subexpr_eliminate.rs

```diff
@@ -829,6 +849,213 @@ fn replace_common_expr(
     .data()
 }
 
+struct ProjectionAdder {
+    // Keeps track of cumulative usage of common expressions with its corresponding data type.
+    // accross plan where key is unsafe nodes that cumulative tracking is invalidated.
+    insertion_point_map: HashMap<usize, HashMap<Expr, (DataType, u32)>>,
+    depth: usize,
+    // Keeps track of cumulative usage of the common expressions with its corresponding data type.
+    // between safe nodes.
+    complex_exprs: HashMap<Expr, (DataType, u32)>,
+}
+pub fn is_not_complex(op: &Operator) -> bool {
+    matches!(
+        op,
+        &Operator::Eq | &Operator::NotEq | &Operator::Lt | &Operator::Gt | &Operator::And
+    )
+}
+
+impl ProjectionAdder {
+    // TODO: adding more expressions for sub query, currently only support for Simple Binary Expressions
```

Comment:
Is there a tracking issue for this that we could refer to in the comment?

Reply:
I'll add that